injector/injector.go

```diff
@@ -0,0 +1,238 @@
+func (m *Mutator) HandleMutate(w http.ResponseWriter, r *http.Request) {
+	log.Println("Handling mutate request, injecting sgx tolerations")
+	body := checkRequest(w, r)
+	if body == nil {
+		// Error was already written to w
+		return
+	}
+
+	// mutate the request and add sgx tolerations to pod
+	mutatedBody, err := mutate(body, m.CoordAddr, true)
+	if err != nil {
+		http.Error(w, "unable to mutate request", http.StatusInternalServerError)
+		return
+	}
+
+	w.WriteHeader(http.StatusOK)
+	w.Write(mutatedBody)
+}
+
+// HandleMutateNoSgx is called when the sgx injection label is not set
+func (m *Mutator) HandleMutateNoSgx(w http.ResponseWriter, r *http.Request) {
+	log.Println("Handling mutate request, omitting sgx injection")
+	body := checkRequest(w, r)
+	if body == nil {
+		// Error was already written to w
+		return
+	}
+
+	// mutate the request and add sgx tolerations to pod
+	mutatedBody, err := mutate(body, m.CoordAddr, false)
+	if err != nil {
+		http.Error(w, "unable to mutate request", http.StatusInternalServerError)
+		return
+	}
+
+	w.WriteHeader(http.StatusOK)
+	w.Write(mutatedBody)
+}
+
+// mutate handles the creation of json patches for pods
+func mutate(body []byte, coordAddr string, injectSgx bool) ([]byte, error) {
+	admReviewReq := v1.AdmissionReview{}
+	if err := json.Unmarshal(body, &admReviewReq); err != nil {
+		return nil, errors.New("invalid admission review")
+	}
+
+	if admReviewReq.Request == nil {
+		return nil, errors.New("empty admission request")
+	}
+
+	var pod corev1.Pod
+	if err := json.Unmarshal(admReviewReq.Request.Object.Raw, &pod); err != nil {
+		return nil, errors.New("invalid pod")
+	}
+
+	// admission response
+	admReviewResponse := v1.AdmissionReview{
+		TypeMeta: metav1.TypeMeta{
+			Kind:       "AdmissionReview",
+			APIVersion: "admission.k8s.io/v1",
+		},
+		Response: &v1.AdmissionResponse{
+			UID: admReviewReq.Request.UID,
+		},
+	}
+
+	pT := v1.PatchTypeJSONPatch
+	admReviewResponse.Response.PatchType = &pT
+
+	// get marble type from pod labels
+	marbleType := pod.Labels["marblerun.marbletype"]
+	// reject pod if label does not exist
+	if len(marbleType) == 0 {
+		admReviewResponse.Response.Allowed = false
+		admReviewResponse.Response.Result = &metav1.Status{
+			Status: "Rejected",
+			Reason: "Missing required label: [marblerun.marbletype]",
+		}
+		bytes, err := json.Marshal(admReviewResponse)
+		if err != nil {
+			return nil, errors.New("unable to marshal admission response")
+		}
+		return bytes, nil
+	}
+
+	// get namespace of pod
+	namespace := pod.Namespace
+	if len(namespace) == 0 {
+		namespace = "default"
+	}
+
+	newEnvVars := []corev1.EnvVar{
+		{
+			Name:  "EDG_MARBLE_COORDINATOR_ADDR",
+			Value: coordAddr,
+		},
+		{
+			Name:  "EDG_MARBLE_TYPE",
+			Value: marbleType,
+		},
+		{
+			Name:  "EDG_MARBLE_DNS_NAMES",
+			Value: fmt.Sprintf("%s,%s.%s,%s.%s.svc.cluster.local", marbleType, marbleType, namespace, marbleType, namespace),
+		},
+		{
+			Name: "EDG_MARBLE_UUID_FILE",
```

Comment:
One more thing. I missed this in my previous review, sorry.
We can omit actually changing this behavior by mounting the information as a file and pointing the env var to this file:
* https://kubernetes.io/docs/concepts/storage/volumes/#downwardapi
* https://kubernetes.io/docs/tasks/inject-data-application/downward-api-volume-expose-pod-information/